src/SFML/Graphics/Shader.cpp

```diff
@@ -156,11 +210,50 @@ Shader::CurrentTextureType Shader::CurrentTexture;
 
 
 ////////////////////////////////////////////////////////////
+struct Shader::UniformBinder : private sf::NonCopyable
+{
+    ////////////////////////////////////////////////////////////
+    /// \brief Constructor: set up state before uniform is set
+    ///
+    ////////////////////////////////////////////////////////////
+    UniformBinder(Shader* shader, const std::string& name) :
+    program(0),
+    location(-1)
+    {
+        if (shader->m_shaderProgram)
```

Comment:
I wanted to reduce code duplication as much as possible. I already lost a line due to Laurent, but at one point it's enough! :angry: :stuck_out_tongue:

No really, I still think having a `UniformBinder` declaration in the header is the least of all evils, definitely better than boilerplate code duplication. It's not part of the interface (because private), and it won't change and lead to recompilations.